### PR TITLE
[out_stackdriver]: Add generic resource keys.

### PR DESCRIPTION
**Pull request by heartrobotninja**
Added keys for generic_node and generic_task.
- location
- namespace
- node_id
- job
- task_id

Signed-off-by: Joey DeStefanis <jdestefanis@google.com>

### CODE REVIEW

File: pipeline/outputs/stackdriver.md [hunk 1]
```diff
@@ -16,7 +16,12 @@ Before to get started with the plugin configuration, make sure to obtain the pro
...
 | metadata\_server | Prefix for a metadata server. Can also set environment variable _$METADATA\_SERVER_.  | http://metadata.google.internal |
-| resource | Set resource type of data. Supported resource types: _k8s\_container_, _k8s\_node_, _k8s\_pod_, _global_ and _gce\_instance_. | global, gce\_instance |
+| resource | Set resource type of data. Supported resource types: _k8s\_container_, _k8s\_node_, _k8s\_pod_, _global_, _generic\_node_, _generic\_task_, and _gce\_instance_. | global, gce\_instance |
+| location | The GCP or AWS region in which to store data about the resource. Used for _generic\_node_ and _generic\_task_.|  |
+| namespace | A namespace identifier, such as a cluster name or environment. Used for _generic\_node_ and _generic\_task_.|  |
+| node_id | A unique identifier for the node within the namespace, such as hostname or IP address. Used for _generic\_node_.|  |
+| job | An identifier for a grouping of related task, such as the name of a  microservice or distributed batch. Used for _generic\_task_.|  |
```
Comment (igorpeshansky) [inline pipeline/outputs/stackdriver.md:23]: Nit: an extra space in "a  microservice".

File: pipeline/outputs/stackdriver.md [hunk 2]
```diff
@@ -16,7 +16,12 @@ Before to get started with the plugin configuration, make sure to obtain the pro
 | service\_account\_email | Account email associated to the service. Only available if **no credentials file** has been provided. | Value of environment variable _$SERVICE\_ACCOUNT\_EMAIL_ |
 | service\_account\_secret | Private key content associated with the service account. Only available if **no credentials file** has been provided. | Value of environment variable _$SERVICE\_ACCOUNT\_SECRET_ |
 | metadata\_server | Prefix for a metadata server. Can also set environment variable _$METADATA\_SERVER_.  | http://metadata.google.internal |
-| resource | Set resource type of data. Supported resource types: _k8s\_container_, _k8s\_node_, _k8s\_pod_, _global_ and _gce\_instance_. | global, gce\_instance |
+| resource | Set resource type of data. Supported resource types: _k8s\_container_, _k8s\_node_, _k8s\_pod_, _global_, _generic\_node_, _generic\_task_, and _gce\_instance_. | global, gce\_instance |
+| location | The GCP or AWS region in which to store data about the resource. Used for _generic\_node_ and _generic\_task_.|  |
```
Comment (igorpeshansky) [inline pipeline/outputs/stackdriver.md:20]: Should there be a space between the period after `_generic\_task_.` and the closing `|`? Also below.

File: pipeline/outputs/stackdriver.md [hunk 3]
```diff
@@ -16,7 +16,12 @@ Before to get started with the plugin configuration, make sure to obtain the pro
...
+| namespace | A namespace identifier, such as a cluster name or environment. Used for _generic\_node_ and _generic\_task_.|  |
```
Comment (igorpeshansky) [inline pipeline/outputs/stackdriver.md:21]: Do you want to match the language in the description for `k8s\_cluster\_name`, etc, i.e.: `If the resource type is one of _generic\_node_ or _generic\_task_, then this field is required.`? Also for other fields.

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: pipeline/outputs/stackdriver.md (hunk 1 above)
Comment (heartrobotninja) [reply pipeline/outputs/stackdriver.md:23]: Done.

File: pipeline/outputs/stackdriver.md (hunk 2 above)
Comment (heartrobotninja) [reply pipeline/outputs/stackdriver.md:20]: Fixed.

File: pipeline/outputs/stackdriver.md (hunk 3 above)
Comment (heartrobotninja) [reply pipeline/outputs/stackdriver.md:21]: Yeah, Fixed.

File: pipeline/outputs/stackdriver.md [hunk 4]
```diff
@@ -16,7 +16,12 @@ Before to get started with the plugin configuration, make sure to obtain the pro
 | service\_account\_email | Account email associated to the service. Only available if **no credentials file** has been provided. | Value of environment variable _$SERVICE\_ACCOUNT\_EMAIL_ |
 | service\_account\_secret | Private key content associated with the service account. Only available if **no credentials file** has been provided. | Value of environment variable _$SERVICE\_ACCOUNT\_SECRET_ |
 | metadata\_server | Prefix for a metadata server. Can also set environment variable _$METADATA\_SERVER_.  | http://metadata.google.internal |
-| resource | Set resource type of data. Supported resource types: _k8s\_container_, _k8s\_node_, _k8s\_pod_, _global_ and _gce\_instance_. | global, gce\_instance |
+| resource | Set resource type of data. Supported resource types: _k8s\_container_, _k8s\_node_, _k8s\_pod_, _global_, _generic\_node_, _generic\_task_, and _gce\_instance_. | global, gce\_instance |
+| location | The GCP or AWS region in which to store data about the resource. If the resource type is one of the  _generic\_node_ and _generic\_task_, then this field is required. |  |
```
Comment (igorpeshansky) [inline pipeline/outputs/stackdriver.md:20]: `_generic\_node_ or _generic\_task_`

**Review by igorpeshansky — CHANGES_REQUESTED** [review]

File: pipeline/outputs/stackdriver.md (hunk 4 above)
Comment (heartrobotninja) [reply pipeline/outputs/stackdriver.md]: Fixed.

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Pull request merged**